src/Screens/Main/Settings/Email.tsx

```diff
@@ -0,0 +1,214 @@
+import React from 'react';
+import RN from 'react-native';
+import * as redux from 'redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { SafeAreaView } from 'react-navigation';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as RD from '@devexperts/remote-data-ts';
+import * as O from 'fp-ts/lib/Option';
+
+import * as navigationProps from '../../../lib/navigation-props';
+
+import * as changeEmailState from '../../../state/reducers/changeEmail';
+import * as accountState from '../../../state/reducers/userAccount';
+import * as actions from '../../../state/actions';
+
+import Message from '../../components/Message';
+import Button from '../../components/Button';
+import { textShadow } from '../../components/shadow';
+import ScreenTitle from '../../components/ScreenTitle';
+import colors from '../../components/colors';
+import fonts from '../../components/fonts';
+import CreatedBySosBanner from '../../components/CreatedBySosBanner';
+import Spinner from '../../components/Spinner';
+
+import { MentorListRoute } from '../../Onboarding/MentorList';
+import NamedInputField from '../../components/NamedInputField';
+
+import AlertBox from './UserAccount/AlertBox';
+
+export type EmailChangeRoute = {
+  'Main/Settings/EmailChange': {};
+};
+
+type Props = navigationProps.NavigationProps<EmailChangeRoute, MentorListRoute>;
+
+export default ({ navigation }: Props) => {
+  const account = RD.toOption(useSelector(accountState.getAccount));
+  const storedEmail = pipe(
+    account,
+    O.map(({ email }) => email),
+    O.toUndefined,
+  );
+  const storedUsername = pipe(
+    account,
+    O.map(({ userName }) => userName),
+    O.toUndefined,
+  );
+
+  const [email, setEmail] = React.useState(storedEmail || '');
+  const dispatch = useDispatch<redux.Dispatch<actions.Action>>();
+
+  const onGoBack = () => {
+    navigation.goBack();
+  };
+  const onButtonPress = () => {
+    dispatch(
+      changeEmailState.changeEmail({ email, account: O.toUndefined(account) }),
+    );
+  };
+  const requestState = useSelector(changeEmailState.select);
+
+  React.useEffect(() => {
+    if (RD.isSuccess(requestState)) {
+      setEmail(requestState.value.email || '');
+      const timeout = setTimeout(onGoBack, changeEmailState.coolDownDuration);
+      return () => clearTimeout(timeout);
+    }
+  }, [requestState]);
+  return (
```

Comment:
[nitpicky] good to have one empty line before return statement. This should be a prettier / eslint rule.